crates/sui/src/client_ptb/parser.rs

```diff
@@ -643,16 +646,46 @@ impl<'a, I: Iterator<Item = &'a str>> ProgramParser<'a, I> {
         })
     }
 
-    /// Parse a numerical or named address.
+    fn parse_mvr_address(&mut self, prefix: Option<&str>) -> PTBResult<Spanned<ParsedAddress>> {
+        use Lexeme as L;
+        use Token as T;
+
+        let mut address = String::new();
+        if let Some(prefix) = prefix {
+            address.push_str(prefix);
+        }
+        let start_sp = self.peek().span;
+        let mut token = self.peek();
+
+        loop {
+            if let sp!(_, L(T::ColonColon, _)) = token {
+                break;
+            }
+            address.push_str(token.value.1);
+            self.bump();
+            token = self.peek();
+        }
+
+        let end_sp = self.peek().span;
+        self.state.mvr_names.insert(address.clone());
+        Ok(start_sp.widen(end_sp).wrap(ParsedAddress::Named(address)))
+    }
+
+    /// Parse a numerical, named address, or mvr address.
     fn parse_address(&mut self) -> PTBResult<Spanned<ParsedAddress>> {
         use Lexeme as L;
         use Token as T;
 
         let sp!(sp, lexeme) = self.peek();
         let addr = match lexeme {
+            L(T::At, _) => self.parse_mvr_address(None)?.value,
             L(T::Ident, name) => {
                 self.bump();
-                ParsedAddress::Named(name.to_owned())
+                if let sp!(_, L(T::At | T::Dot | T::ForwardSlash, _)) = self.peek() {
```

Comment:
This should be made stricter. I think you should only look for a `T::Dot` case here. e.g., this is the `foo.sui` case so no `@` or `/` should be found